subsys/bluetooth/mesh/adv_multi.c

```diff
@@ -0,0 +1,393 @@
+	};
+	uint16_t duration, adv_int;
+	struct bt_data ad;
+	int err;
+
+	adv_int = MAX(ADV_INT_FAST_MS,
+		      BT_MESH_TRANSMIT_INT(BT_MESH_ADV(buf)->xmit));
+	/* Upper boundary estimate: */
+	duration = start.num_events * (adv_int + 10);
+
+	BT_DBG("type %u len %u: %s", BT_MESH_ADV(buf)->type,
+	       buf->len, bt_hex(buf->data, buf->len));
+	BT_DBG("count %u interval %ums duration %ums",
+	       BT_MESH_TRANSMIT_COUNT(BT_MESH_ADV(buf)->xmit) + 1, adv_int,
+	       duration);
+
+	ad.type = bt_mesh_adv_type[BT_MESH_ADV(buf)->type];
+	ad.data_len = buf->len;
+	ad.data = buf->data;
+
+	/* Only update advertising parameters if they're different */
+	if (adv_param.interval_min != BT_MESH_ADV_SCAN_UNIT(adv_int)) {
+		adv_param.interval_min = BT_MESH_ADV_SCAN_UNIT(adv_int);
+		adv_param.interval_max = adv_param.interval_min;
+	}
+
+	err = adv_start(adv_ins, &adv_param, &start, &ad, 1, NULL, 0);
+	if (!err) {
+		adv_ins->buf = net_buf_ref(buf);
+	}
+
+	bt_mesh_adv_send_start(duration, err, BT_MESH_ADV(buf));
+
+	return err;
+}
+
+static void send_pending_adv(struct k_work *work)
+{
+	struct ext_adv *adv_ins = CONTAINER_OF(work, struct ext_adv, work.work);
+	struct net_buf *buf;
+	int err;
+
+	atomic_clear_bit(adv_ins->flags, ADV_FLAG_SCHEDULED);
+
+	while ((buf = net_buf_get(adv_ins->fifo, K_NO_WAIT))) {
+		/* busy == 0 means this was canceled */
+		if (!BT_MESH_ADV(buf)->busy) {
+			net_buf_unref(buf);
+			continue;
+		}
+
+		BT_MESH_ADV(buf)->busy = 0U;
+		err = buf_send(adv_ins, buf);
+
+		net_buf_unref(buf);
+
+		if (!err) {
+			return; /* Wait for advertising to finish */
+		}
+	}
+}
+
+static void send_gatt_adv(struct k_work *work)
+{
+	struct ext_adv *adv_ins = BT_MESH_GATT_ADV_INS;
+
+	if (atomic_test_and_clear_bit(adv_ins->flags, ADV_FLAG_ACTIVE)) {
+		(void)bt_le_ext_adv_stop(adv_ins->instance);
+	}
+
+	if (bt_mesh_is_provisioned()) {
+		if (IS_ENABLED(CONFIG_BT_MESH_GATT_PROXY)) {
+			(void)bt_mesh_proxy_adv_start();
+			BT_DBG("Proxy Advertising");
+		}
+	} else if (IS_ENABLED(CONFIG_BT_MESH_PB_GATT)) {
+		(void)bt_mesh_pb_gatt_adv_start();
+		BT_DBG("PB-GATT Advertising");
+	}
+}
+
+static void schedule_local_send(void)
+{
+	struct ext_adv *adv_ins = BT_MESH_LOCAL_ADV_INS;
+	uint64_t timestamp = adv_ins->timestamp;
+	int64_t delta;
+
+	if (atomic_test_bit(adv_ins->flags, ADV_FLAG_ACTIVE) ||
+	    atomic_test_and_set_bit(adv_ins->flags, ADV_FLAG_SCHEDULED)) {
+		return;
+	}
+
+	/* The controller will send the next advertisement immediately.
+	 * Introduce a delay here to avoid sending the next mesh packet closer
+	 * to the previous packet than what's permitted by the specification.
+	 */
+	delta = k_uptime_delta(&timestamp);
+	k_work_reschedule(&adv_ins->work, K_MSEC(ADV_INT_FAST_MS - delta));
+}
+
+static void schedule_relay_send(void)
+{
+	int64_t delta;
+	struct ext_adv *adv_ins = BT_MESH_RELAY_ADV_INS_START;
+
+	if (!adv_ins) {
+		return;
+	}
+
+	for (; adv_ins <= BT_MESH_RELAY_ADV_INS_END; adv_ins++) {
```

Comment:
```suggestion
	for (; adv_ins <= (struct ext_adv*)BT_MESH_RELAY_ADV_INS_END; adv_ins++) {
```

<warning: comparison of distinct pointer types lacks a cast>

Reply:
Is it necessary? Or did you encounter this warning?

Reply:
Got this warning while compiling the changes

Reply:
OK. thanks, i will rebase.